consensus/types/src/beacon_state/iter.rs

```diff
@@ -108,46 +109,47 @@ mod test {
             .map(|i| (Slot::from(i), *state.get_block_root(Slot::from(i)).unwrap()))
             .collect::<Vec<_>>();
         assert_eq!(
-            all_roots(&state, &spec),
+            all_roots(&state),
             expected,
             "slot higher than the block roots history"
         );
     }
 
-    #[test]
-    fn block_roots_iter_non_zero_genesis() {
-        let mut spec = E::default_spec();
-        spec.genesis_slot = Slot::new(4);
-
-        let mut state: BeaconState<E> = BeaconState::new(0, <_>::default(), &spec);
-
-        for i in 0..state.block_roots().len() {
-            *state.block_roots_mut().get_mut(i).unwrap() = root_slot(i).1;
-        }
-
-        assert_eq!(
-            state.slot(),
-            spec.genesis_slot,
-            "test assume a genesis slot state"
-        );
-        assert_eq!(
-            all_roots(&state, &spec),
-            vec![],
-            "state at genesis slot has no history"
-        );
-
-        *state.slot_mut() = Slot::new(5);
-        assert_eq!(
-            all_roots(&state, &spec),
-            vec![root_slot(4)],
-            "first slot after genesis has one slot history"
-        );
-
-        *state.slot_mut() = Slot::new(6);
-        assert_eq!(
-            all_roots(&state, &spec),
-            vec![root_slot(5), root_slot(4)],
-            "second slot after genesis has two slot history"
-        );
-    }
+    // TODO(do we need this test?)
```

Comment:
I think you can revive this test if instead of using `all_roots` you directly contruct an iter: `BlockRootsIter::new(self, GENESIS_SLOT)`